### PR TITLE
Windows fixes

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
While batch files normally work with LFs, they are subtly broken when using labels and goto (which ours use).

- Ensure that batch files always get checked in and out of git with CRLFs
- Add the ability to specify a specific Python interpreter on Windows when running the superbuild

These issues don't show up on our existing maxquant build system due to the way git and python are configured, but this fixes the build on our replacement systems and end user systems with different global git configuration.

Testing: check out this branch.  Delete scripts/jenkins-build.bat and reset your working tree. Run `file` on the recreated file and it should have CRLFs.

### CODE REVIEW

**Comment by rleigh-codelibre** [comment]
- Ensure that batch files always get checked in and out of git with CRLFs
- Add the ability to specify a specific Python interpreter on Windows when running the superbuild

These issues don't show up on our existing maxquant build system due to the way git and python are configured, but this fixes the build on our replacement systems and end user systems with different global git configuration.

Testing: check out this branch.  Delete scripts/jenkins-build.bat and reset your working tree. Run `file` on the recreated file and it should have CRLFs.

**Comment by dgault** [comment]
Not sure what has gone wrong but I'm not seeing the `scripts/jenkins-build.bat` with CRLFs unfortunately. The `scripts/jenkins-build` file does have CRLFs but not the batch file. This is testing on a Windows machine, deleting the file and then resetting the branch.

Any ideas on what might have happened?

**Comment by rleigh-codelibre** [comment]
Depends on your local git config, and which commit you're on when you reset the branch.  The Windows git config is a bit odd, the installer options overriding the system config.  Try

```
git config --system -l
git config --global -l
git config --local -l
```

the last inside the repo.

It might be worth trying a fresh clone of the repo and then checking out this branch, in case the line endings are "sticky".  Then see if it changed, then delete and reset.

Also, try on plain unix; should work on both but it may be the git installer registry settings are overriding the local file attributes.

**Comment by sbesson** [comment]
@dgault does this need to be relisted for re-reivew?

**Comment by dgault** [comment]
Yeah I will need to re-review this, added it to tomorrows list

File: .gitattributes [hunk 1]
```diff
@@ -1 +1,3 @@
 *.diff -text
+*.bat text eol=crlf
+*.bat text eol=lf
```
Comment (dgault) [inline .gitattributes]: Is .bat supposed to have 2 entries for eol?

Comment (rleigh-codelibre) [reply .gitattributes]: Yes.  The first is for checkout, and the second is for checkin.  I can only assume this was a retrofit to a format which originally used one entry for specifying both.  However, while several stackoverflow posts recommend this, it's not reflected in the documentation.  So it's either undocumented, or wrong (but harmless).  Seemed to work during my Windows and Unix testing.

**Comment by rleigh-codelibre** [comment]
I've pushed a further change restricting it to CRLF only in the gitattributes file.  It seems to work fine with a checkout on Windows and Linux.

**Comment by mtbc** [comment]
Without this PR:
```
scripts/jenkins-build.bat: DOS batch file, ASCII text
```
With this PR:
```
scripts/jenkins-build.bat: DOS batch file, ASCII text, with CRLF line terminators
```
:+1:

**Pull request merged**